packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { koFetchMock, okFetchMock } from '../__mocks__/fetch.mock';
+import { fetchHttpClient } from '../fetch.http-client';
+
+describe('fetch httpClient testing', () => {
+  const endpoint = 'https://api.empathy.co/search';
+
+  beforeEach(() => {
+    window.fetch = okFetchMock as any;
+    jest.clearAllMocks();
+  });
+
+  it('creates well formed and valid URLs', async () => {
+    await fetchHttpClient(endpoint, {
+      parameters: {
+        q: 'shirt',
+        filter: ['long sleeve', 'dotted', 'white'],
+        rows: 12
+      }
+    });
+    expectFetchCall(`${endpoint}?q=shirt&filter=long+sleeve&filter=dotted&filter=white&rows=12`);
+  });
+
+  it('allows to pass headers to the request', () => {
+    const headers = { instance: 'A1B1' };
+    fetchHttpClient(endpoint, {
+      properties: {
+        headers
+      }
+    });
+
+    expect(window.fetch).toBeCalledWith(endpoint, { headers, signal: expect.anything() });
+  });
+
+  it('allows URLs which already have parameters', async () => {
+    await fetchHttpClient(`${endpoint}?additionalParam=true`, {
+      parameters: {
+        q: 'shirt'
+      }
+    });
+    expectFetchCall(`${endpoint}?additionalParam=true&q=shirt`);
+  });
+
+  it('does not map undefined values', async () => {
+    await fetchHttpClient(endpoint, {
+      parameters: {
+        q: undefined
+      }
+    });
+    expectFetchCall(endpoint);
+  });
+
+  it('cancels equal endpoint requests if no requestId parameter is passed', async () => {
+    expect.hasAssertions();
```

Comment:
A little more specific with: `expect.assertions(2);`

Reply:
You're right. I'll refactor those.